### PR TITLE
Fix flaky shared arbitrator tests

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Concurrent arbitrator test discovered some race condition in hash build operators
that can lead to wrong task failure. For instance, when the last hash build operator
failed to build the table, it will leave the hash join bridge in a inconsistent state that
cause the other build operators fail when reach to the join bridge. Since these are
running in parallel, it can lead to incorrect the task failure if the other hash build
operators set the task error first. With a temporal fix, the test can easily get into stuck.
This PR disable the concurrent test and re-enable after all the fixes are in.
This PR also simplify an existing test case.

### CODE REVIEW

File: velox/exec/tests/SharedArbitratorTest.cpp [hunk 1]
```diff
@@ -669,6 +668,29 @@ class SharedArbitrationTest : public exec::test::HiveConnectorTestBase {
     return result;
   }
 
+  QueryTestResult runFakeTask(
```
Comment (bikramSingh91) [inline velox/exec/tests/SharedArbitratorTest.cpp:671]: where is this used?

File: velox/exec/tests/SharedArbitratorTest.cpp [hunk 2]
```diff
@@ -1839,132 +1861,22 @@ TEST_F(SharedArbitrationTest, reclaimFromCompletedJoinBuilder) {
...
-                        .planNode())
-              .assertResults("SELECT * FROM tmp");
-    });
-    joinThread.join();
-    memThread.join();
-    waitForAllTasksToBeDeleted();
-    ASSERT_GT(arbitrator_->stats().numRequests, 0);
-  }
+  // Create a query ctx with a small capacity to trigger spilling.
+  std::shared_ptr<core::QueryCtx> queryCtx = newQueryCtx(128 << 20);
```
Comment (bikramSingh91) [inline velox/exec/tests/SharedArbitratorTest.cpp:1870]: just to clarify, does this mean that any attempt to spill would go via the arbitrator?

File: velox/exec/tests/SharedArbitratorTest.cpp (hunk 1 above)
Comment (xiaoxmeng) [reply velox/exec/tests/SharedArbitratorTest.cpp:671]: I will use it in followup for fake memory task.

File: velox/exec/tests/SharedArbitratorTest.cpp (hunk 2 above)
Comment (xiaoxmeng) [reply velox/exec/tests/SharedArbitratorTest.cpp:1870]: We have threshold based spilling trigger but we don't leverage this in shared arbitrator test. In shared arbitrator test, the spilling is always triggered from arbitrator.

**Pull request closed without merging**